clients/src/main/java/org/apache/kafka/common/requests/ShareFetchRequest.java

```diff
@@ -41,7 +41,7 @@ public static class Builder extends AbstractRequest.Builder<ShareFetchRequest> {
         private final ShareFetchRequestData data;
 
         public Builder(ShareFetchRequestData data) {
-            this(data, false);
+            this(data, true);
```

Comment:
We should not do this. Rather construct the request in the code by using the other constructor.